### PR TITLE
ToDo App by Linnéa Helén

### PR DESCRIPTION
**Pull request by Polkapetunian**
Link to my deployed project: https://linneas-todo-app.netlify.app/

My reflections on how this week's project turned out: I am satisfied with the result. Red goals achieved. 

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/package.json [hunk 1]
```diff
@@ -3,16 +3,22 @@
   "version": "1.0.0",
   "private": true,
   "dependencies": {
+    "@reduxjs/toolkit": "^1.5.1",
     "babel-eslint": "^10.0.3",
+    "dayjs": "^1.10.4",
```
Comment (aicb1) [inline code/package.json]: Well done installing more dependencies!

File: code/public/index.html [hunk 2]
```diff
@@ -13,7 +13,8 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>Technigo React App</title>
+    <title>To Do App</title>
```
Comment (aicb1) [inline code/public/index.html:16]: Good job remembering to change the title to reflect the theme of your app 👍

File: code/src/App.js [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
+const store = configureStore({ reducer});
+
+const AppContainer = styled.div`
+  width: 320px;
+  height: 100vh;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: center;
+  @media (min-width: 768px) {
```
Comment (aicb1) [inline code/src/App.js:26]: Your media query works great, nice practicing this with styled components!

File: code/src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+display: flex;
+flex-direction: column;
+font-family: 'Noto Sans KR', sans-serif;
+@media (min-width: 768px){
+  font-size: 22px;
+}
+`;
+const Header1 = styled.h1`
+margin: 0;
+font-family: 'Rammetto One', cursive;
```
Comment (aicb1) [inline code/src/components/Header.js]: When loading the app in the browser the h1 is set to cursive for a few seconds then changes. Not sure if this is intentional or if you wanted to stick to one font style only ?

File: code/src/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+const Header = () => {
+  return (
+    <Container>
+      <div>
+        <Header1>ToDo</Header1>
+        <DateContainer>
+          <DayJS format="YYYY-MM-DD"/>
+          <DayJS format="HH:mm"/>
+        </DateContainer>
+
```
Comment (aicb1) [inline code/src/components/Header.js]: The space could be removed for convention ? But nice indenting!

File: code/src/components/TaskCard.js [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+align-items: center;
+@media (min-width: 768px){
+  width: 600px;
+}
+`;
+const Label = styled.label`
+display: inline-block;
+position: relative;
+padding-left: 35px;
+margin: 0px;
```
Comment (aicb1) [inline code/src/components/TaskCard.js]: px could be removed for 0 margin values ?

File: code/src/components/TaskCard.js [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+          {isComplete && 
+          <Tick src="./assets/tick-sign.svg" alt="tick"/>}
+        </Label>
+        <Button onClick={onClickAction}>
+          <Wastebin src="./assets/wastebin.svg"/>
+        </Button>
+      </UpperRow>
+    <CreatedAt>
+      <SpanDay>Added </SpanDay>
+      <DayJS element="span" format="YYYY-MM-DD H:mm">{createdAt}</DayJS>
```
Comment (aicb1) [inline code/src/components/TaskCard.js:155]: Great dates. Well done! 😀

File: code/src/components/TaskCreator.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+width: 25px; 
+height: 25px;
+border-radius: 50%;
+background-color: #2f5d62;
+position: relative;
+border: none;
+&:hover { background-color: #6c9fa3;
+cursor: pointer;}
+&:active { background-color: #ffb037;}
+&:disabled {opacity: 0.6;
```
Comment (aicb1) [inline code/src/components/TaskCreator.js]: Nice disabling of the button for control !

**Review by aicb1 — APPROVED** [review]
Your app works great, have managed to add and delete todos. 

Very nice icons used as well! 🎉

File: code/src/components/TaskCreator.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+margin: 10px 0;
+display: flex;
+flex-direction: row;
+justify-content: space-evenly;
+background-color: white;
+border-radius: 15px;
+box-shadow: 0  0 10px #aaaaaa;
+@media (min-width: 768px){
+  width: 680px;
+}
```
Comment (KarolCislowski) [inline code/src/components/TaskCreator.js]: Why you have no indentation in your style-components?

File: code/src/components/ToDoList.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+const Container = styled.div`
+width: 100%;
+display; flex;
+flex-direction: column;
+justify-content: center;
+align-items: center;
+@media (min-width: 768px){
+  width: 680px;
+}
+`;
```
Comment (KarolCislowski) [inline code/src/components/ToDoList.js:18]: remember to separate your container with 1 empty line. It is easier to read 👓

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Great job 🦄  but please provide indentation inside of your styled component and fix tasks text (wrap or max-length on input)

File: code/src/components/Header.js (hunk 4 above)
Comment (Polkapetunian) [reply code/src/components/Header.js]: I tried replacing the fallback font to another more similar to Rammetto One, but not sure it works...

File: code/src/components/Header.js (hunk 5 above)
Comment (Polkapetunian) [reply code/src/components/Header.js]: Fixed

File: code/src/components/TaskCard.js (hunk 6 above)
Comment (Polkapetunian) [reply code/src/components/TaskCard.js]: Fixed

File: code/src/components/TaskCreator.js (hunk 9 above)
Comment (Polkapetunian) [reply code/src/components/TaskCreator.js]: Fixed. Did not know how the should look.

File: code/src/components/ToDoList.js (hunk 10 above)
Comment (Polkapetunian) [reply code/src/components/ToDoList.js:18]: Fixed.

**Comment by Polkapetunian** [comment]
I have made changes and hope it's enough to pass.  =)

**Review by KarolCislowski — APPROVED** [review]
now it works great!

**Pull request closed without merging**